File: Dddml.Wms.Common/Specialization/ApplicationContext.cs
using System;


namespace Dddml.Wms.Specialization
{
    public abstract class ApplicationContext : IApplicationContext
    {

        private static volatile ApplicationContext _current;

        public static ApplicationContext Current
        {
            get
            {
                return _current;
            }
            set
            {
                _current = value;
            }
        }

        private ITypeConverter _typeConverter = new DefaultTypeConverter();

        public virtual ITypeConverter TypeConverter
        {
            get { return _typeConverter; }
            set { _typeConverter = value; }
        }

        public abstract object this[string name]
        {
            get;
        }

    }
}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/specialization/ApplicationContext.java
package org.dddml.wms.specialization;


/**
 * Created by Yang on 2016/7/20.
 */
public class ApplicationContext {

    public static volatile ApplicationContext current;

    public static void setCurrent(ApplicationContext context) {
        current = context;
    }

    //private ITypeConverter _typeConverter = new DefaultTypeConverter();

    public Object get(String name) {
        throw new UnsupportedOperationException();
    }

}
